### PR TITLE
feat: Overhaul Java documentation

### PR DESCRIPTION
**Pull request by dcramer**
- Support dynamic versions when available
- Remove invalid instructions on index - you need to use an integration
- Move manual usage under core Java (its the same as a guide)
- Add DSN placeholders
- Similar tweaks to Android
- Modernize Gradle instructions

### CODE REVIEW

File: src/platforms/android/index.mdx [hunk 1]
```diff
@@ -31,7 +31,7 @@ android {
 // ADD SENTRY ANDROID AS A DEPENDENCY
 dependencies {
     // https://github.com/getsentry/sentry-android/releases
-    implementation 'io.sentry:sentry-android:{version}'
+    implementation 'io.sentry:sentry-android:{{ packages.version('sentry.android', '{version}') }}'
```
Comment (dcramer) [inline src/platforms/android/index.mdx:34]: @bruno-garcia fyi made an assumption on what the package name will be called, same w/ sentry.java below

**Comment by dcramer** [comment]
image

image

File: src/platforms/java/guides/google-app-engine/index.mdx [hunk 2]
```diff
@@ -12,29 +12,23 @@ The source can be found [on GitHub](https://github.com/getsentry/sentry-java/tre
...
+    <version>{{ packages.version('sentry.java', '1.7.30') }}</version>
 </dependency>
 ```
 
-Using Gradle:
-
-```groovy
-compile 'io.sentry:sentry-appengine:1.7.30'
+```groovy {tabTitle:Gradle}
+compile 'io.sentry:sentry-appengine:{{ packages.version('sentry.java', '1.7.30') }}'
```
Comment (bruno-garcia) [inline src/platforms/java/guides/google-app-engine/index.mdx:24]:
```suggestion
implementation 'io.sentry:sentry-appengine:{{ packages.version('sentry.java', '1.7.30') }}'
```

File: src/platforms/java/guides/log4j/index.mdx [hunk 3]
```diff
@@ -10,29 +10,23 @@ The source can be found [on GitHub](https://github.com/getsentry/sentry-java/tre
...
+    <version>{{ packages.version('sentry.java', '1.7.30') }}</version>
 </dependency>
 ```
 
-Using Gradle:
-
-```groovy
-compile 'io.sentry:sentry-log4j:1.7.30'
+```groovy {tabTitle:Gradle}
+compile 'io.sentry:sentry-log4j:{{ packages.version('sentry.java', '1.7.30') }}'
```
Comment (bruno-garcia) [inline src/platforms/java/guides/log4j/index.mdx:22]:
```suggestion
implementation 'io.sentry:sentry-log4j:{{ packages.version('sentry.java', '1.7.30') }}'
```

File: src/platforms/java/guides/log4j2/index.mdx [hunk 4]
```diff
@@ -10,29 +10,23 @@ The source can be found [on GitHub](https://github.com/getsentry/sentry-java/tre
...
+    <version>{{ packages.version('sentry.java', '1.7.30') }}</version>
 </dependency>
 ```
 
-Using Gradle:
-
-```groovy
-compile 'io.sentry:sentry-log4j2:1.7.30'
+```groovy {tabTitle:Gradle}
+compile 'io.sentry:sentry-log4j2:{{ packages.version('sentry.java', '1.7.30') }}'
```
Comment (bruno-garcia) [inline src/platforms/java/guides/log4j2/index.mdx:22]:
```suggestion
implementation 'io.sentry:sentry-log4j2:{{ packages.version('sentry.java', '1.7.30') }}'
```

File: src/platforms/java/guides/logback/index.mdx [hunk 5]
```diff
@@ -10,29 +10,23 @@ The source can be found [on GitHub](https://github.com/getsentry/sentry-java/tre
...
+    <version>{{ packages.version('sentry.java', '1.7.30') }}</version>
 </dependency>
 ```
 
-Using Gradle:
-
-```groovy
-compile 'io.sentry:sentry-logback:1.7.30'
+```groovy {tabTitle:Gradle}
+compile 'io.sentry:sentry-logback:{{ packages.version('sentry.java', '1.7.30') }}'
```
Comment (bruno-garcia) [inline src/platforms/java/guides/logback/index.mdx:22]:
```suggestion
implementation 'io.sentry:sentry-logback:{{ packages.version('sentry.java', '1.7.30') }}'
```

File: src/platforms/java/guides/logging/index.mdx [hunk 6]
```diff
@@ -10,29 +10,23 @@ The source for `sentry` can be found [on GitHub](https://github.com/getsentry/se
...
+    <version>{{ packages.version('sentry.java', '1.7.30') }}</version>
 </dependency>
 ```
 
-Using Gradle:
-
-```groovy
-compile 'io.sentry:sentry:1.7.30'
+```groovy {tabTitle:Gradle}
+compile 'io.sentry:sentry:{{ packages.version('sentry.java', '1.7.30') }}'
```
Comment (bruno-garcia) [inline src/platforms/java/guides/logging/index.mdx:22]:
```suggestion
implementation 'io.sentry:sentry:{{ packages.version('sentry.java', '1.7.30') }}'
```

File: src/platforms/java/guides/spring/index.mdx [hunk 7]
```diff
@@ -8,29 +8,23 @@ The source can be found [on GitHub](https://github.com/getsentry/sentry-java/tre
...
+    <version>{{ packages.version('sentry.java', '1.7.30') }}</version>
 </dependency>
 ```
 
-Using Gradle:
-
-```groovy
-compile 'io.sentry:sentry-spring:1.7.30'
+```groovy {tabTitle:Gradle}
+compile 'io.sentry:sentry-spring:{{ packages.version('sentry.java', '1.7.30') }}'
```
Comment (bruno-garcia) [inline src/platforms/java/guides/spring/index.mdx:20]:
```suggestion
implementation 'io.sentry:sentry-spring:{{ packages.version('sentry.java', '1.7.30') }}'
```

File: src/platforms/java/usage.mdx [hunk 8]
```diff
@@ -1,35 +1,29 @@
...
     <artifactId>sentry</artifactId>
     <version>1.7.30</version>
 </dependency>
 ```
 
-Using Gradle:
-
-```groovy
+```groovy {tabTitle:Gradle}
 compile 'io.sentry:sentry:1.7.30'
```
Comment (bruno-garcia) [inline src/platforms/java/usage.mdx:19]:
```suggestion
implementation 'io.sentry:sentry:1.7.30'
```

File: src/wizard/java/appengine.md [hunk 9]
```diff
@@ -13,24 +13,22 @@ Using Maven:
...
-    <version>1.7.30</version>
+    <version>{{ packages.version('sentry.java', '1.7.30') }}</version>
 </dependency>
 ```
 
 Using Gradle:
 
 ```groovy
-compile 'io.sentry:sentry-appengine:1.7.30'
+compile 'io.sentry:sentry-appengine:{{ packages.version('sentry.java', '1.7.30') }}'
```
Comment (bruno-garcia) [inline src/wizard/java/appengine.md:23]:
```suggestion
implementation 'io.sentry:sentry-appengine:{{ packages.version('sentry.java', '1.7.30') }}'
```

File: src/wizard/java/index.md [hunk 10]
```diff
@@ -13,24 +13,22 @@ Using Maven:
...
-    <version>1.7.30</version>
+    <version>{{ packages.version('sentry.java', '1.7.30') }}</version>
 </dependency>
 ```
 
 Using Gradle:
 
 ```groovy
-compile 'io.sentry:sentry:1.7.30'
+compile 'io.sentry:sentry:{{ packages.version('sentry.java', '1.7.30') }}'
```
Comment (bruno-garcia) [inline src/wizard/java/index.md:23]:
```suggestion
implementation 'io.sentry:sentry:{{ packages.version('sentry.java', '1.7.30') }}'
```

File: src/wizard/java/log4j.md [hunk 11]
```diff
@@ -13,24 +13,22 @@ Using Maven:
...
-    <version>1.7.30</version>
+    <version>{{ packages.version('sentry.java', '1.7.30') }}</version>
 </dependency>
 ```
 
 Using Gradle:
 
 ```groovy
-compile 'io.sentry:sentry-log4j:1.7.30'
+compile 'io.sentry:sentry-log4j:{{ packages.version('sentry.java', '1.7.30') }}'
```
Comment (bruno-garcia) [inline src/wizard/java/log4j.md:23]:
```suggestion
implementation 'io.sentry:sentry-log4j:{{ packages.version('sentry.java', '1.7.30') }}'
```

File: src/wizard/java/log4j2.md [hunk 12]
```diff
@@ -13,24 +13,22 @@ Using Maven:
...
-    <version>1.7.30</version>
+    <version>{{ packages.version('sentry.java', '1.7.30') }}</version>
 </dependency>
 ```
 
 Using Gradle:
 
 ```groovy
-compile 'io.sentry:sentry-log4j2:1.7.30'
+compile 'io.sentry:sentry-log4j2:{{ packages.version('sentry.java', '1.7.30') }}'
```
Comment (bruno-garcia) [inline src/wizard/java/log4j2.md:23]:
```suggestion
implementation 'io.sentry:sentry-log4j2:{{ packages.version('sentry.java', '1.7.30') }}'
```

File: src/wizard/java/logback.md [hunk 13]
```diff
@@ -13,24 +13,22 @@ Using Maven:
...
-    <version>1.7.30</version>
+    <version>{{ packages.version('sentry.java', '1.7.30') }}</version>
 </dependency>
 ```
 
 Using Gradle:
 
 ```groovy
-compile 'io.sentry:sentry-logback:1.7.30'
+compile 'io.sentry:sentry-logback:{{ packages.version('sentry.java', '1.7.30') }}'
```
Comment (bruno-garcia) [inline src/wizard/java/logback.md:23]:
```suggestion
implementation 'io.sentry:sentry-logback:{{ packages.version('sentry.java', '1.7.30') }}'
```

File: src/wizard/java/logging.md [hunk 14]
```diff
@@ -13,24 +13,22 @@ Using Maven:
...
-    <version>1.7.30</version>
+    <version>{{ packages.version('sentry.java', '1.7.30') }}</version>
 </dependency>
 ```
 
 Using Gradle:
 
 ```groovy
-compile 'io.sentry:sentry:1.7.30'
+compile 'io.sentry:sentry:{{ packages.version('sentry.java', '1.7.30') }}'
```
Comment (bruno-garcia) [inline src/wizard/java/logging.md:23]:
```suggestion
implementation 'io.sentry:sentry:{{ packages.version('sentry.java', '1.7.30') }}'
```

**Pull request merged**